### PR TITLE
webrender: render basic text

### PR DESCRIPTION
**Pull request by harryfei**

### CODE REVIEW

**Comment by yyoncho** [comment]
Do you expect something to be tested or only a code review?

**Comment by harryfei** [comment]
@yyoncho Only code review. I'm trying to merge https://github.com/harryfei/emacs-ng/commits/webrender-lib branch by multiple smaller PRs.

File: rust_src/src/vector_macros.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+macro_rules! offset_of {
```
Comment (DavidDeSimone) [inline rust_src/src/vector_macros.rs:1]: I recall problems with offset_of in remacs: https://github.com/remacs/remacs/issues/1564. I don't think the situation has changed in terms of there being no ready solution yet, however I think we should still track it. It should not block your PR, just wanted to point it out.

**Comment by brotzeit** [comment]
Sorry I got bad timing with #62. I will do the rebasing.

**Comment by brotzeit** [comment]
@harryfei got it soon.

**Comment by brotzeit** [comment]
I start today with updating this branch.

**Comment by brotzeit** [comment]
@harryfei the remaining errors are mostly

```
error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
```

I think you know better how we want to modify the code that it will work again.

**Comment by harryfei** [comment]
@brotzeit I will recreate a PR based on changes in this PR.

**Comment by brotzeit** [comment]
@harryfei sure

**Comment by harryfei** [comment]
Updated.

**Review by brotzeit — APPROVED** [review]

**Pull request merged**